lib/Backend/LowerMDShared.cpp

```diff
@@ -8745,17 +8758,18 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
                 Assert(src->IsFloat32());
                 zero = IR::MemRefOpnd::New((float*)&Js::JavascriptNumber::k_Float32Zero, TyFloat32, this->m_func, IR::AddrOpndKindDynamicFloatRef);
             }
+
+            IR::LabelInstr * skipRoundSd = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
             if(instr->m_opcode == Js::OpCode::InlineMathRound)
             {
+                IR::LabelInstr* addHalfToRoundSrcLabel = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
+                IR::LabelInstr* skipAddHalfLabel = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
+
                 if(instr->ShouldCheckForNegativeZero())
```

Comment:
If this evaluates to false, we would generate wrong code for numbers < 0.5. Code generated by `this->m_lowerer->InsertCompareBranch(roundedFloat, pointFive, Js::OpCode::BrLt_A, skipAddHalfLabel, instr);` will always jump to skipAddHalfLabel